cluster-autoscaler/core/dynamic_autoscaler.go

```diff
@@ -34,12 +34,16 @@ type DynamicAutoscaler struct {
 }
 
 // NewDynamicAutoscaler builds a DynamicAutoscaler from required parameters
-func NewDynamicAutoscaler(autoscalerBuilder AutoscalerBuilder, configFetcher dynamic.ConfigFetcher) *DynamicAutoscaler {
+func NewDynamicAutoscaler(autoscalerBuilder AutoscalerBuilder, configFetcher dynamic.ConfigFetcher) (*DynamicAutoscaler, error) {
+	autoscaler, err := autoscalerBuilder.Build()
+	if err != nil {
+		return nil, err
+	}
 	return &DynamicAutoscaler{
-		autoscaler:        autoscalerBuilder.Build(),
+		autoscaler:        autoscaler,
 		autoscalerBuilder: autoscalerBuilder,
 		configFetcher:     configFetcher,
-	}
+	}, err
```

Comment:
I know err == nil if we get here, but I think it would more intuitive if you return nil directly here (I started thinking what the err value is reading it)